### ISSUE TITLE
Curriculum expert dropout

### ISSUE THREAD

**Issue by kyo219** [kyo219/LightGBM-MoE#15]
## Problem

Expert dropout rate is fixed throughout training (`mixture_gbdt.cpp:1682-1718`):

```cpp
if (dropout_dist_(dropout_rng_) < dropout_rate) {
    expert_active[k] = false;
}
```

A fixed rate is suboptimal — early training benefits from low dropout (let all experts learn), while later training benefits from higher dropout (stronger regularization to prevent co-adaptation).

## Proposed Solution

Schedule dropout rate as a function of training progress:

```
dropout(t) = dropout_min + (dropout_max - dropout_min) * (t / T)
```

New parameters:
- `mixture_expert_dropout_schedule`: "constant" (current) | "linear" | "cosine"
- `mixture_expert_dropout_min`: starting rate
- `mixture_expert_dropout_max`: ending rate

## Expected Impact

3-5% better generalization, reduced expert co-adaptation.

## Files

- `src/boosting/mixture_gbdt.cpp` (lines 1682-1718)
- `include/LightGBM/config.h` (new parameters)
